File: src/apis/ui-api/http-client.js
const DEFAULT_HEADERS = {
  "Content-Type": "application/json",
  "Accept-Encoding": "gzip, deflate, br",
};

class HttpClient {
  constructor(baseUrl) {
    this.BASE_URL = baseUrl;
  }

  async get(path, headers = {}) {
    const url = `${this.BASE_URL}/${path}`;
    const response = await fetch(url, {
      method: "GET",
      headers: {
        ...DEFAULT_HEADERS,
        ...headers,
      },
    });
    if (!response.ok) {
      throw new Error(
        `GET request to ${url} failed with status ${response.status}`
      );
    }
    return response.json();
  }

  async post(path, data, headers = {}) {
    const url = `${this.BASE_URL}/${path}`;
    const response = await fetch(url, {
      method: "POST",
      headers: {
        ...DEFAULT_HEADERS,
        ...headers,
      },
      body: JSON.stringify(data),
    });

    if (!response.ok) {
      throw new Error(
        `POST request to ${url} failed with status ${response.status}`
      );
    }

    return response.json();
  }
}

module.exports = HttpClient;

File: src/modules/common/src/pm_common/http_clients/http_client.py
from typing import Optional
import requests


class HttpClient:
    _default_headers = {
        "Content-Type": "application/json",
        "Accept-Encoding": "gzip, deflate, br",
    }

    def __init__(self, base_url: Optional[str] = ""):
        self._base_url = base_url

    def post(
        self,
        json: str,
        url: Optional[str] = "",
        headers: Optional[dict[str, str]] = _default_headers,
    ):
        return requests.request(
            "POST", url=f"{self._base_url}/{url}", headers=headers, data=json
        )
